pommerman/network/ion_client/__init__.py

```diff
@@ -0,0 +1,219 @@
+        ui.info(ui.yellow, Strings.server_comm.value)
+    try:
+        network.join_list(room)
+    except Exception as e:
+        if ui_en:
+            ui.fatal(e)
+        raise e
+    if ui_en:
+        ui.info(Strings.match_variant.value, ui.yellow, network.mode)
+        ui.info(ui.yellow, Strings.match_wait.value)
+    try:
+        network.waitm()
+    except Exception as e:
+        if ui_en:
+            ui.fatal(e)
+        raise e
+    if ui_en:
+        ui.info(Strings.match_run.value, "#" + network.match_id)
+    while (True):
+        try:
+            match_obj = network.match_get()
+        except Exception as e:
+            if ui_en:
+                ui.fatal(e)
+            raise e
+        if match_obj[0] is 0:
+            action = agent.act(match_obj[1], Discrete(6))
+            try:
+                network.send_move(action, match_obj[2])
+            except Exception as e:
+                if ui_en:
+                    ui.fatal(e)
+                raise e
+        elif match_obj[0] is 2:
+            agent.episode_end(match_obj[1])
+            if match_obj[1] == 1:
+                ui.info(Strings.match_won.value)
+            if match_obj[1] == -1:
+                ui.info(Strings.match_loss_draw.value)
+            ui.info(
+                Strings.match_agent.value, ui.yellow,
+                Item(match_obj[2]).name
+            )
+            break
+    ui.info(Strings.match_replay.value, ui.yellow, network.match_id)
+    if ui.ask_yes_no(Strings.match_ask_replay.value):
+        replay(network, network.match_id)
+    else:
+        intent(network)
+
+
+def replay(network, id=False, ui_en=True):
+    """Arguments:  
+    * network: An `network`(pommerman.network.ion_client.network) object  
+    * id: The ID of a match to be played. If False, the user is prompted about \
+it.  
+    * ui_en: If the UI is enabled or disabled (This also controls if exception are\
+raised or not)  
+    Description: This replays a particular match"""
+    if not id and ui_en:
+        id = str(ui.ask_string(Strings.replay_prompt.value))
+        if id is None:
+            ui.fatal(Strings.error_invalid_id.value)
+    ui.info(
+        Strings.server_replay_p1.value, ui.yellow, "#" + str(id), ui.reset,
+        Strings.server_replay_p2.value
+    )
+    try:
+        r = network.get_replay(id)
+    except Exception as e:
+        if ui_en:
+            ui.fatal(e)
+        raise e
+    if ui_en:
+        ui.info(Strings.replay_start.value, ui.yellow, "#" + str(id))
+    # Note: Imports are local as they aren't required anywhere else
+    from pommerman import make
+    from pommerman.agents import BaseAgent
+    from numpy import array
+    env = make(r["mode"], [BaseAgent(), BaseAgent(), BaseAgent(), BaseAgent()])
+    env.reset()
+    env._board = array(r["board"])
+    # Note: Render FPS is set to 30 as it'll be smoother
+    env._render_fps = 30
+    for i in r["actions"]:
+        env.render()
+        rew = env.step(i)[1]
```

Comment:
Please stop abbreviating. It's really unnecessary and is at best distracting, at worst confusing.

Reply:
Agreed, it's a bad habit 👍